components/tooltip/base.ts

```diff
@@ -116,13 +114,21 @@ export abstract class NzTooltipBaseDirective implements OnChanges, OnDestroy, Af
 
   private delayTimer?: ReturnType<typeof setTimeout>;
 
-  constructor(
-    public elementRef: ElementRef,
-    protected hostView: ViewContainerRef,
-    protected renderer: Renderer2,
-    protected noAnimation?: NzNoAnimationDirective,
-    protected nzConfigService?: NzConfigService
-  ) {}
+  elementRef = inject(ElementRef);
+  protected hostView = inject(ViewContainerRef);
+  protected renderer = inject(Renderer2);
+  protected noAnimation = inject(NzNoAnimationDirective, { host: true, optional: true });
+  protected nzConfigService = inject(NzConfigService);
+  protected platformId = inject(PLATFORM_ID);
+
+  constructor(protected componentType: Type<NzTooltipBaseComponent>) {}
+
+  ngAfterViewInit(): void {
+    if (isPlatformBrowser(this.platformId)) {
+      this.createComponent();
+      this.registerTriggers();
+    }
+  }
```

Comment:
The reason for the hydration failure is that the host view node was removed using the native API when the component was dynamically created, resulting in the node not being found when hydrating.

While the issue can be resolved by using `ViewContainerRef.detach` instead of removeChild, this also results in components within the overlay not being able to correctly measure element sizes.

By controlling this part of the logic to be executed only in the client, we can temporarily solve this problem.

In conclusion, we should not remove the host view of a component after it has been rendered. The best solution here would be to use the `Overlay.create` API to render the component directly to the overlay container so that there is no need to `removeChild`.